batch/src/main/java/org/openmrs/analytics/FhirEtl.java

```diff
@@ -157,8 +157,8 @@ static void runFhirJdbcFetch(FhirEtlOptions options, FhirContext fhirContext)
 		    options.getTableFhirMapPath());
 		// process each table-resource mappings
 		for (Map.Entry<String, String> entry : reverseMap.entrySet()) {
-			String tableName = entry.getKey();
-			String resourceType = entry.getValue();
+			String tableName = entry.getValue();
+			String resourceType = entry.getKey();
 			String baseBundleUrl = options.getOpenmrsServerUrl() + options.getServerFhirEndpoint() + "/" + resourceType;
 			int maxId = jdbcUtil.fetchMaxId(tableName);
```

Comment:
I think you can make this code more efficient by avoiding `fetchMaxId` multiple times for the same table. For example if we have both `Patient` and `Person` in the `searchList`, we will end up finding max ID twice for the `person` table. Same goes for ID-range generation (next line) which can be done once for each table.

Reply:
done